### PR TITLE
Pavel working on register user

### PR DESCRIPTION
**Pull request by 2HEADSs**
fix register functionality to work with real post requst and return token. Remove hardcore data and already fixed comments

### CODE REVIEW

File: src/models/User.js [hunk 1]
```diff
@@ -36,37 +35,36 @@ const userSchema = new Schema({
...
-    maxlength: [15, "Role must be maximum 15 characters long!"],
-    */
+    enum: ["user", "admin"],
+    default: "user"
   },
-  city: { type: String, required: true },
-  /* TODO: Propper validation for city length
+  city: {
+    type: String,
+    required: true,
```
Comment (Getsov) [inline src/models/User.js:43]: city for user is not required

Comment (2HEADSs) [reply src/models/User.js]: already fixed and pushed

File: src/services/userService.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ const ObjectId = mongoose.Types.ObjectId;
 //TODO: use env and change secret
 const secret = "q213fdfsddfasd231adfas12321kl";
 
-    async function registerUser(email, firstName, lastName,city, address, password) {
+    async function registerUser(email, firstName, lastName, password) {
```
Comment (Getsov) [inline src/services/userService.js:10]: add the city to be optional, don't delete it

File: src/services/userService.js [hunk 3]
```diff
@@ -18,8 +18,6 @@ const secret = "q213fdfsddfasd231adfas12321kl";
             email,
             firstName,
             lastName,
-            city,
```
Comment (Getsov) [inline src/services/userService.js:26]: add the city to be optional

File: src/controllers/userController.js [hunk 4]
```diff
@@ -5,13 +5,19 @@ const { registerUser, loginUser } = require('../services/userService');
...
+        const userData = {
+            email: req.body.email,
+            firstName: req.body.firstName,
+            lastName: req.body.lastName,
+            password: req.body.password,
+        };
+
+        if (req.body.city) {
+            userData.city = req.body.city;
+        }
```
Comment (Getsov) [inline src/controllers/userController.js:19]: else -> userData.city = "";

File: src/services/userService.js [hunk 5]
```diff
@@ -7,24 +7,23 @@ const ObjectId = mongoose.Types.ObjectId;
...
+  const existing = await User.findOne({ email });
+  if (existing) {
+    throw new Error("Email is already taken!!!");
+  }
 
+  const user = await User.create({
+    email: userData.email,
+    firstName: userData.firstName,
+    lastName: userData.lastName,
+    city: userData.city ? userData.city : "",
```
Comment (Getsov) [inline src/services/userService.js:21]: city: userData.city

**Review by Getsov — APPROVED** [review]

**Pull request merged**